components/match2/wikis/counterstrike/match_group_input_custom.lua

```diff
@@ -0,0 +1,692 @@
+		mapveto = matchFunctions.getMapVeto(match),
+		status = matchFunctions.getMatchStatus(match),
+		overturned = Logic.isNotEmpty(match.overturned),
+		featured = matchFunctions.isFeatured(match),
+		hidden = Logic.readBool(Variables.varDefault('match_hidden'))
+	}
+	return match
+end
+
+-- Parse the mapVeto input
+function matchFunctions.getMapVeto(match)
+	if not match.mapveto then return nil end
+
+	match.mapveto = Json.parseIfString(match.mapveto)
+
+	local vetotypes = mw.text.split(match.mapveto.types or '', ',')
+	local deciders = mw.text.split(match.mapveto.decider or '', ',')
+	local vetostart = match.mapveto.firstpick or ''
+	local deciderIndex = 1
+
+	local data = {}
+	for index, vetoType in ipairs(vetotypes) do
+		vetoType = mw.text.trim(vetoType):lower()
+		if not Table.includes(ALLOWED_VETOES, vetoType) then
+			return nil -- Any invalid input will not store (ie hide) all vetoes.
+		end
+		if vetoType == 'decider' then
+			table.insert(data, {type = vetoType, decider = deciders[deciderIndex]})
+			deciderIndex = deciderIndex + 1
+		else
+			table.insert(data, {type = vetoType, team1 = match.mapveto['t1map'..index], team2 = match.mapveto['t2map'..index]})
+		end
+	end
+	if data[1] then
+		data[1].vetostart = vetostart
+	end
+	return data
+end
+
+function matchFunctions.getOpponents(match)
+	-- read opponents and ignore empty ones
+	local opponents = {}
+	local isScoreSet = false
+	for opponentIndex = 1, MAX_NUM_OPPONENTS do
+		-- read opponent
+		local opponent = match['opponent' .. opponentIndex]
+		if not Logic.isEmpty(opponent) then
+			CustomMatchGroupInput.processOpponent(opponent, match.date)
+
+			-- Retrieve icon for team
+			if opponent.type == Opponent.team then
+				opponent.icon, opponent.icondark = opponentFunctions.getIcon(opponent.template)
+			end
+
+			-- apply status
+			if TypeUtil.isNumeric(opponent.score) then
+				opponent.status = 'S'
+				isScoreSet = true
+			elseif Table.includes(ALLOWED_STATUSES, opponent.score) then
+				opponent.status = opponent.score
+				opponent.score = -1
+			end
+			opponents[opponentIndex] = opponent
+
+			-- get players from vars for teams
+			if opponent.type == 'team' and not Logic.isEmpty(opponent.name) then
+				match = matchFunctions.getPlayers(match, opponentIndex, opponent.name)
+			end
+		end
+	end
+
+	-- Handle tournament status for unfinished matches
+	if (not Logic.readBool(match.finished)) and Logic.isNotEmpty(match.status) then
+		match.finished = match.status
+	end
+
+	if Table.includes(NP_MATCH_STATUS, match.finished) then
+		match.resulttype = 'np'
+		match.status = match.finished
+		match.finished = true
+	else
+		-- see if match should actually be finished if score is set
+		if isScoreSet and not Logic.readBool(match.finished) and match.hasDate then
+			local currentUnixTime = os.time(os.date('!*t'))
+			local lang = mw.getContentLanguage()
+			local matchUnixTime = tonumber(lang:formatDate('U', match.date))
+			local threshold = match.dateexact and 30800 or 86400
```

Comment:
imo push the numbers up top

Reply:
It's fine imo, most other wikis does this and I'm planning on getting the timestamp rework done and will be touching this across all wikis